packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -95,18 +103,17 @@ export abstract class AbstractAsyncHooksContextManager
    * @param context the context we want to bind
    * @param ee EventEmitter an instance of EventEmitter to patch
    */
-  private _bindEventEmitter<T extends EventEmitter>(
-    context: Context,
-    ee: T
-  ): T {
+  private _bindEventEmitter(context: Context, ee: EventEmitter): EventEmitter {
```

Comment:
Accepting `<T extends EventEmitter>` causes the assignment below to require a cast without really adding much in type safety.

The reason is that if the input is `T extends EventEmitter`, technically, the signature for the `ee[methodName]` can be different than `EventEmitter[methodName]`, so that assignment won't work. To make that work requires propagating the generic to everywhere else which adds a lot of noise with little benefit (ultimately it will boil down to an unsafe cast involving exactly this issue anyway).